### PR TITLE
DOC: Add a test layout subsection.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add a test layout subsection to develop the layout of unit tests in ITK.

Change-Id: I9734932cf63bacab48d433a0b139b8bfaa584ce3

### CODE REVIEW

File: SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex [hunk 1]
```diff
@@ -3484,6 +3484,24 @@ \section{Writing Tests}
 in ITK.
 
 
+\subsection{Code Layout in Tests}
+\label{subsec:CodeLayoutInTests}
+
+The following general layout is recommended for ITK unit tests:
+\begin{itemize}
+\item Constant declarations (e.g. image dimensions, etc.).
+\item General typedef declarations (e.g. pixel and image types, etc.).
```
Comment (thewtex) [inline SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex]: It helps readability to have constants and typedef's local to where they are used -- we could leave these two items out.

File: SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex [hunk 2]
```diff
@@ -3484,6 +3484,24 @@ \section{Writing Tests}
...
+\item Input image read (or generation).
+\item \code{foo} class instantiation and basic object checks (e.g.
+\code{EXERCISE\_BASIC\_OBJECT\_METHODS}).
+\item \code{foo} class properties' input argument read and test (e.g. using the
+macros in \code{itkTestingMacro.h}, such as \code{TEST\_SET\_GET\_OBJECT}, etc.).
+\item \code{foo} class update.
```
Comment (thewtex) [inline SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex]: update -> \code{Update()}

Comment (jhlegarreta) [reply SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex]: Done in 08df219

File: SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex (hunk 1 above)
Comment (jhlegarreta) [reply SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex]: 08df219 removes the item but adds a note after the item list. Have a look to see if that makes sense or whether I should rather drop the note and only leave the items out.

**Comment by jhlegarreta** [comment]
08df219 addresses the comments by @thewtex and was rebased on master.

**Pull request merged**